lib/bridge.js

```diff
@@ -17,51 +18,111 @@ var Callback = require('./callback');
  */
 var Bridge = function Bridge (ip, port, token, options) {
   if (!(this instanceof Bridge)) {
-    return new Bridge(ip, port, token);
+    return new Bridge(ip, port, token, options);
   }
 
   this.ip = ip;
   this.port = parseInt(port, 10);
   this.token = token;
+  this.type = 1; // 1 = hardware, 2 = software
+  
   this.delayBetweenRequests = options && options.delayBetweenRequests || 250;
   this.delayer = Promise.resolve();
 
   if (!this.ip || !this.port || !this.token || this.port < 1 || this.port > 65536) {
     throw new Error('Please check the arguments!');
   }
+  
+  // get bridge type
+  this._getBridgeType();
+};
+
+/**
+ * This function retrieves the type of bridge (1 = hardware, 2 = software).
+ *
+ * @param void
+ * @returns void
+ * @private
+ */
+Bridge.prototype._getBridgeType = function _getBridgeType () {
+
+	var self = this;
+	self.delayer = new Promise(resolve =>
```

Comment:
Can you please change the { on the next line to be on the same as the arrow? I think it's ok to use es6 features and bump the required node version to 8, since it's near before the EOL